server/config/config.go

```diff
@@ -613,6 +570,49 @@ func (c *Config) Adjust(meta *toml.MetaData, reloading bool) error {
 	return nil
 }
 
+// Utility to test if a configuration is defined.
+type configMetaData struct {
+	meta *toml.MetaData
+	path []string
```

Comment:
This `path` is kind of confusing to me. Does it mean the config file path or the internal config key path? What about using `keyPath`?

Reply:
configMetaData is an existing structure and the issue is a legacy issue. This PR just moves configMetaData from other place without modifying anything except the place, so I think we can address @JmPotato's concern in separate PR.